src/EditorFeatures/Core/InlineHints/InlineHintsDataTaggerProvider.cs

```diff
@@ -93,6 +93,9 @@ protected override async Task ProduceTagsAsync(TaggerContext<InlineHintDataTag>
         {
             var cancellationToken = context.CancellationToken;
             var document = documentSnapshotSpan.Document;
+            if (document == null)
+                return;
```

Comment:
Nice!

Reply:
(although this makes me wonder why the base async tagger doesn't filter out things not attached to a document, but that can be for another day)

Reply:
it's weird.  it looks intentional as hte base tagger explicit has a note saying it is letting this through.  i imagine we must have some tagger that still works without a doc.  That said, i would prefer to maybe filter those out an dhave derived types opt into stating that they don't support document-less tagging.

Reply:
Agreed, it definitely looks intentional, but I can't explain why at this point. We'd have to research further.